esp/services/ws_workunits/ws_workunitsService.cpp

```diff
@@ -3313,6 +3313,31 @@ void getClusterConfig(char const * clusterType, char const * clusterName, char c
     return;
 }
 
+bool CWsWorkunitsEx::onWUGraphQuery(IEspContext &context,IEspWUGraphQueryRequest &req, IEspWUGraphQueryResponse &resp)
```

Comment:
Actually it's not as bad as I thought... getGraphsMeta IS lightweight(ish) but the additional information you fill in beyond the name and type makes it more expensive - e.g. retrieving the graph state requires a fetch into the progress information and looking up the start/end times requires a read of the wu statistics. If you were JUST fetching names (as originally suggested) then commoning up with the WuInfo graph fetch would not really make sense.

Listing the response type as being ECLGraph in the scm file is misleading if this is only partially filled in - you should change it to indicate what is actually returned.